### DISCUSSION TITLE
Full Modal (Text Input) Support

### DISCUSSION THREAD

**Discussion by hassiexx** [bwmarrin/discordgo#1139] [Ideas]
Using the library today, I noticed it does not fully support creating modals using message components. Currently the library only contains TextInput struct for the text input options. However, to fully support modals, an outer object is required to set the title and ID of the modal.

From Discord API:
```json
{
  "title": "My Cool Modal",
  "custom_id": "cool_modal",
  "components": [{
    "type": 1,
    "components": [{
      "type": 4,
      "custom_id": "name",
      "label": "Name",
      "style": 1,
      "min_length": 1,
      "max_length": 4000,
      "placeholder": "John",
      "required": true
    }]
  }]
}
```

**Comment by hassiexx** [comment]
Created an issue instead.

**Reply by FedorLap2006** [reply]
Just a tip,
![image](https://user-images.githubusercontent.com/42876964/159814030-ba7d8e80-a789-4e3d-b5d3-3b3f864c24af.png)
could be used to automatically transform a discussion into an issue
